design-documents/graph-ql/coverage/customer-orders.md

```diff
@@ -79,55 +80,40 @@ The `id` will be a `base64_encode(increment_id)` which in future can be replaced
 
 ### Order Item
 
-The order items will be presented as separate interface which will have multiple implementations for invoice, shipment and credit memo types.
-
 ```graphql
-@doc("Interface to reprent order/invoice/shipment/credit memo items")
-interface SalesItemInterface {
+@doc("Order item")
+type OrderItemInterface {
+    id: ID! @doc("Order item unique identifier") #base64encode(orderItemId)
     product_name: String @doc("name of the base product")
     product_sku: String! @doc("SKU of the base product")
-    product_url: String @doc("URL of the base product")
+    product_url_key: String @doc("URL key of the base product")
+    product_type: String @doc("Type of product (e.g. simple, configurable, bundle)")
```

Comment:
Discussed offline, recording here for posterity:

@danielrenaud and I discussed a few things about `product_type` offline:

1. We talked about a similar field being deprecated recently in core, [`ProductInterface.type_id`](https://github.com/magento/magento2/blob/7f2272dcbaccde66519f7d1b3c81a633ff539b3c/app/code/Magento/CatalogGraphQl/etc/schema.graphqls#L108), with the recommendation to use `__typename` at runtime. We're not returning the full entity type here though (since order items are a limited, historical snapshot of the item), so that doesn't make much sense.

1. We discussed the possibility of making this an `enum`, rather than the wider `String` type. That would help a client understand what values to handle, but it sounds like Magento doesn't have a static list of product types (is extensible).